### PR TITLE
adding vendor progression

### PR DESCRIPTION
**Pull request by kyleshay**
screen shot 2016-05-15 at 3 02 38 pm

Hover gives you the name and actual numbers.

Implementation is a bit sloppy. Can easily move some thing around.

The call to get the progress of the vendors is asynchronously done after the page is actually loaded, so that the initial page load is still the same.

Maybe it would be better to implement the progression as "items" so that they have a popup dialog similar to other things, and will properly show up under a `dimCategory`, versus being hard-coded to render in the `dimStoreItems.directive.js`.

### CODE REVIEW

**Comment by kyleshay** [comment]
how they look w/ 3 and 5 columns:

screen shot 2016-05-15 at 3 09 43 pm
screen shot 2016-05-15 at 3 09 30 pm

**Comment by kyleshay** [comment]
solves #521, could easily #254 with this as well. wasn't sure if it should show up as a vendor, or should show up in the bar below the characters.

**Comment by SunburnedGoose** [comment]
#254 should show up as a blue bar beneath the character once they have reached max level for that 5 mote drop.

**Comment by kyleshay** [comment]
and done! 💃 

screen shot 2016-05-15 at 3 36 51 pm

**Comment by kyleshay** [comment]
renamed that section to vendors, for now at least... since we already have a section called "progress"

screen shot 2016-05-15 at 3 40 18 pm

**Comment by SunburnedGoose** [comment]
xp 'till 5 motes of light or...

xp until 5 motes of light

**Comment by kyleshay** [comment]
ok i can fix that. seeing a weird issue with the bar after a refresh. clears out... but the function is still returning the right data... don't think we're doing a bind once there... :S

**Comment by SunburnedGoose** [comment]
Shouldn’t be since that data changes on refresh.

**Comment by kyleshay** [comment]
alright, resolved it. happy for people to check it out!!

**Comment by bhollis** [comment]
Super cool!

What would happen if you reduced the opacity of that blue bar?

File: app/scripts/store/dimStoreItems.directive.js [hunk 1]
```diff
@@ -80,6 +80,22 @@
...
+        '      <div class="unequipped">',
+        '        <span class="item" ng-if="faction.color" ng-repeat="faction in vm.store.progression.progressions track by $index" title="{{faction.identifier}}\n{{faction.progressToNextLevel}}/{{faction.nextLevelAt}}">',
+        '          <svg width="48" height="48">',
+        '            <polygon stroke-dasharray="130" fill="{{faction.color}}" points="24,1 47,24 24,47 1,24"/>',
+        '            <image xlink:href="" ng-attr-xlink:href="{{faction.icon | bungieIcon}}" ng-attr-x="{{faction.scale === \'.8\' ? 6 : 48-(faction.scale*48)}}" ng-attr-y="{{faction.scale === \'.8\' ? 6 : 48-(faction.scale*48)}}" width="48" height="48" ng-attr-transform="scale({{faction.scale}})" />',
+        '            <polygon stroke-dasharray="130" fill-opacity="0" stroke="#FFF" stroke-opacity=".35" stroke-width="2" points="24,1 47,24 24,47 1,24"/>',
+        '            <polygon stroke-dasharray="130" style="stroke-dashoffset:{{130-(130*faction.progressToNextLevel/faction.nextLevelAt)}}" fill-opacity="0" stroke="#FFF" stroke-width="2" points="24,1 47,24 24,47 1,24"/>',
+        '          </svg>',
```
Comment (bhollis) [inline app/scripts/store/dimStoreItems.directive.js]: Yasssss more SVG

File: app/scripts/store/dimStoreHeading.directive.js [hunk 2]
```diff
@@ -55,6 +55,18 @@
         });
       });
 
+      vm.getLevelBar = function getLevelBar() {
+        if(vm.store.percentToNextLevel) {
+          return {width: vm.store.percentToNextLevel + '%'};
+        }
+        if(vm.store.progression && vm.store.progression.progressions) {
+          vm.xpTillMote = 'Prestige level: ' + vm.store.progression.progressions[4].level + '\n' +
```
Comment (bhollis) [inline app/scripts/store/dimStoreHeading.directive.js]: It'd be nice if you had a name for this progression rather than having to rely on it being #4.

Comment (kyleshay) [inline app/scripts/store/dimStoreHeading.directive.js]: good call. meant to get back to that. i'll use underscore and pluck via the hash

**Comment by bhollis** [comment]
CHANGELOG?

File: app/scripts/store/dimStoreItems.directive.js (hunk 1 above)
Comment (kyleshay) [inline app/scripts/store/dimStoreItems.directive.js]: don't look too closely... it works but it's... scary >_<

would like the stroke ends of the polygon path to properly intersect...

screen shot 2016-05-16 at 1 12 59 am

File: build/progressionMeta.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (kyleshay) [inline build/progressionMeta.json]: normal progressionDefs from destiny don't give us nice names... maybe it's in another table? either way we need special scales for each one and bg colors...

**Comment by bhollis** [comment]
Rebase and it's good to merge!

**Comment by kyleshay** [comment]
sounds good (fyi @SunburnedGoose is currently doing a release)

**Comment by JosephBono** [comment]
Suggestion - consider naming this section "Reputation" instead of "Vendors". This is the terminology used on Bungie.net and on DestinyStatus.com.

**Comment by JosephBono** [comment]
What are our thoughts for the current ordering of factions in this section (Cryptarch, Crota's Bane, Iron Banner, Vanguard, Crucible, Dead Orbit, Future War Cult, New Monarchy, House of Judgment, Queen's Wrath, Gunsmith, SRL)? What guidelines do we want to use for ordering the factions?

Off the top of my head, the guidelines I would use for coming up with an order are:

- Prioritize factions that provide rank-up packages that users are more likely to be farming
- Put less emphasis on factions that do not provide relevant rewards
- Put less emphasis on factions that are not always available to earn reputation for
- Keep DO, FWC, and NM together
- Keep Vanguard and Crucible together

We don't have a consistent example to work from here:

- In game order: Cryptarch, Vanguard, Crucible, DO/FWC/NM, Crota's Bane, IB, HoJ, QW, Gunsmith, SRL
- Bungie.net order: Vanguard, DO, FWC, NM, Cryptarch, Crucible, SRL, Gunsmith, Crota's Bane, HoJ, QW (unsure where Iron Banner gets inserted)
- DestinyStatus order: Cryptarch, Gunsmith, Vanguard, Crucible, DO, FWC, NM, IB, Crota's Bane, QW, HoJ, SRL

My preference is to either emulate the in game order or come up with our own order that is based on the needs of our users. What do other people think?

**Comment by delphiactual** [comment]
Drag & Drop customizable order by end-user

**Pull request merged**